src/graphql/mutations/password_reset.rs

```diff
@@ -0,0 +1,116 @@
+use async_graphql::{Context, Object};
+use sea_orm::{
+    ActiveValue, ColumnTrait, DatabaseConnection, EntityTrait, IntoActiveModel, QueryFilter,
+};
+use sendgrid::SGClient;
+use tracing::{event, Level};
+
+use crate::{
+    error::new_err,
+    schema::{self, password_reset_tokens, users},
+};
+
+#[derive(Default)]
+pub struct PasswordResetMutation;
+const RESET_URL_BASE: &str = "https://lumina.earth/reset?token=";
+#[Object(rename_fields = "snake_case", rename_args = "snake_case")]
+impl PasswordResetMutation {
+    #[graphql()]
+    pub async fn reset_password(
+        &self,
+        ctx: &Context<'_>,
+        email: String,
+    ) -> async_graphql::Result<&str> {
+        let db = ctx.data_unchecked::<DatabaseConnection>();
+        let s_g_client = ctx.data_unchecked::<SGClient>();
+
+        let user = users::Entity::find()
+            .filter(users::Column::Email.contains(&email))
+            .one(db)
+            .await?
+            .ok_or_else(|| new_err("USER_NOT_FOUND", &format!("User not found: {}", &email)))?;
+
+        let mut new_reset_token = password_reset_tokens::ActiveModel::default();
```

Comment:
I think we should construct the reset token here as a struct and then convert `into_active_model` as it will ensure that future non-nullable fields (expiry, etc) in the insertion are typed & compile-time checked.